--- mongodb-src-r4.4.2/src/mongo/util/errno_util.cpp.orig	2020-12-29 22:49:37.000000000 +1100
+++ mongodb-src-r4.4.2/src/mongo/util/errno_util.cpp	2020-12-29 22:47:35.000000000 +1100
@@ -61,10 +61,7 @@ std::string errnoWithDescription(int err
     char buf[kBuflen];
     char* msg{nullptr};
 
-#if defined(__GNUC__) && defined(_GNU_SOURCE) && \
-    (!defined(__ANDROID_API__) || !(__ANDROID_API__ <= 22)) && !defined(EMSCRIPTEN)
-    msg = strerror_r(errNumber, buf, kBuflen);
-#elif defined(_WIN32)
+#if defined(_WIN32)
 
     LPWSTR errorText = nullptr;
     FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ALLOCATE_BUFFER |
